api/api-access-token-v1/src/main/java/com/thoughtworks/go/apiv1/accessToken/CurrentUserAccessTokenControllerV1.java

```diff
@@ -85,7 +88,7 @@ public String createAccessToken(Request request, Response response) throws Excep
 
         final JsonReader reader = GsonTransformer.getInstance().jsonReaderFrom(request.body());
 
-        String tokenDescription = reader.optString("description").orElse(null);
+        String tokenDescription = reader.getString("description");
```

Comment:
Shouldn't this be a validation error than a 400 response? What if a token is created another way, and somehow a `null` ends up in the DB?

Reply:
@ketan - `reader.getString(...)` returns `422` if `description` is not provided.

```bash
< HTTP/1.1 422 Unprocessable Entity
< Date: Mon, 25 Feb 2019 08:40:42 GMT
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< X-Frame-Options: SAMEORIGIN
< X-UA-Compatible: chrome=1
< Set-Cookie: JSESSIONID=node01lpygaoc8bl43qcv59nuljf1e18.node0;Path=/go;Expires=Mon, 11-Mar-2019 08:40:42 GMT;Max-Age=1209600;HttpOnly
< Expires: Thu, 01 Jan 1970 00:00:00 GMT
< Set-Cookie: JSESSIONID=node011l991wrxtapvtah4mqowdmey19.node0;Path=/go;Expires=Mon, 11-Mar-2019 08:40:42 GMT;Max-Age=1209600;HttpOnly
< Cache-Control: max-age=0, private, must-revalidate
< Content-Type: application/vnd.go.cd.v1+json;charset=utf-8
< X-Runtime: 0
< Transfer-Encoding: chunked
<
* Connection #0 to host localhost left intact
{
  "message": "Json `{\"description1\":\"my first token\"}` does not contain property 'description'"
}
```

Reply:
I'm suggesting that we instead make this a model validation, rather than a controller enforced validation.

Reply:
Make sense. I will revert the change.